packages/react-scripts/scripts/init.js

```diff
@@ -196,21 +224,13 @@ module.exports = function(
   }
 
   // Install additional template dependencies, if present
-  let templateJsonPath;
-  if (templateName) {
-    templateJsonPath = path.join(templatePath, 'template.json');
-  } else {
-    templateJsonPath = path.join(appPath, '.template.dependencies.json');
-  }
-
-  if (fs.existsSync(templateJsonPath)) {
-    const templateDependencies = require(templateJsonPath).dependencies;
+  const templateDependencies = templateJson.dependencies;
+  if (templateDependencies) {
     args = args.concat(
       Object.keys(templateDependencies).map(key => {
         return `${key}@${templateDependencies[key]}`;
       })
     );
-    fs.unlinkSync(templateJsonPath);
```

Comment:
@iansu This may have been what was deleting the `template.json` for some users.

Reply:
Nice catch!